checkbox-provider-ce-oem/bin/tcp_multi_connections.py

```diff
@@ -0,0 +1,340 @@
+    client ports.
+    The server listens on a range of ports, and the clients connect to
+    these ports to send a payload and receive a response from the server.
+
+    Usage:
+    - To run as a server: ./script.py server -p <star_port> -e <end_port>
+    - To run as a client: ./script.py client -H <server_host> -p <start_port>
+      -e <end_port> -P <payload_size>
+
+    Arguments:
+    - mode (str): Specify whether to run as a server or client.
+    - host (str): Server host IP (client mode). This is mandatory arg.
+    - port (int): Starting port for the server or server port for the client.
+      Default is 1024.
+    - payload (int): Payload size in KB for the client. Default is 64.
+    - end_port (int): Ending port for the server. Default is 1223.
+
+    Server Mode:
+    - The server listens on a range of ports concurrently, handling
+      incoming connections and send the received data back to client.
+
+    Client Mode:
+    - The client connects to a range of server ports,
+      sending a payload and validating the received response.
+      The script logs pass, fail, or error status for each port.
+    """
+
+    parser = argparse.ArgumentParser(
+        description="Client-server with payload check on multiple ports")
+
+    subparsers = parser.add_subparsers(dest="mode",
+                                       help="Run as server or client")
+
+    # Subparser for the server command
+    server_parser = subparsers.add_parser("server", help="Run as server")
+    server_parser.add_argument("-p", "--port",
+                               type=int, default=1024,
+                               help="Starting port for the server")
+    server_parser.add_argument("-e", "--end-port", type=int, default=1223,
+                               help="Ending port for the server")
+
+    # Subparser for the client command
+    client_parser = subparsers.add_parser("client", help="Run as client")
+    client_parser.add_argument("-H", "--host", required=True,
+                               help="Server host (client mode)")
+    client_parser.add_argument("-p", "--port", type=int, default=1024,
+                               help="Starting port for the client")
+    client_parser.add_argument("-P", "--payload", type=int, default=64,
+                               help="Payload size in KB (client mode)")
+    client_parser.add_argument("-e", "--end-port", type=int, default=1223,
+                               help="Ending port for the client")
+    args = parser.parse_args()
+
+    results = []
+    # Ramp up time to wait until all ports are connected before
+    # starting to send the payload.
+    start_time = datetime.now() + timedelta(seconds=20)
+
+    if args.mode == "server":
+        server(args.port, args.end_port)
+    elif args.mode == "client":
+        payload = 'A' * (args.payload * 1024)
+        client(args.host, args.port, args.end_port,
+               payload, start_time, results)
```

Comment:
we could just pass the payload length to client function, instead of whole payload data.